src/modules/Game.class.js

```diff
@@ -20,26 +26,358 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(initialState, winCallback, loseCallback) {
+    this.board = initialState || [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.score = 0;
+
+    this.status = IDLE;
+    this.isStarted = false;
+    this.winCallback = winCallback;
+    this.loseCallback = loseCallback;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveTile(fromCell, toCell) {
+    // Отримуємо координати клітинок
+    const fromCellRect = fromCell.getBoundingClientRect();
+    const toCellRect = toCell.getBoundingClientRect();
 
-  /**
-   * @returns {number}
-   */
-  getScore() {}
+    // Вираховуємо зміщення
+    const xOffset = toCellRect.left - fromCellRect.left;
+    const yOffset = toCellRect.top - fromCellRect.top;
+
+    // Встановлюємо зміщення через CSS перемінні
+    fromCell.style.setProperty('--x-offset', `${xOffset}px`);
+    fromCell.style.setProperty('--y-offset', `${yOffset}px`);
+
+    // Додаємо клас для запуску анімації
+    fromCell.classList.add('moving');
+
+    // Затримка для застосування анімації збільшення
+    setTimeout(() => {
+      // Додаємо клас 'merged' для анімації збільшення
+      toCell.classList.add('merged');
+
+      // Затримка для видалення класу 'merged' після завершення анімації
+      setTimeout(() => {
+        toCell.classList.remove('merged');
+      }, 300); // Час анімації (300ms)
```

Comment:
If you change some property in `merged` class you should add transition for this property in scss file
Now it doesn't work (but not required in the task - so it's only a suggestion)